improver/lapse_rate.py

```diff
@@ -455,33 +456,33 @@ def process(self, temperature, orography, land_sea_mask,
 
         # Create list of arrays over "realization" coordinate
```

Comment:
This feels wrong.
What the code is trying to do is to get 2D slices (of x and y) to pass to _generate_lapse_rate_array. The code is assuming that the only other dimension that will ever be present is called "realization".
I think it would be much more sensible to instead use temperature_cube.slices_over("y_name", "x_name") and to build new cubes that inherit all scalar coords from these slices. They will then merge back together at the end. This would allow users to pass in cubes with time coordinates if they chose to arrange their data in that way.

Reply:
Yes.

This may not be quite right.  However I've not changed what this plugin does, and this is something we're not very consistent about throughout the code base.  On discussion we've agreed to leave this for now, and fix it if and when any issues arise.